sourcegraph/sourcegraph
# WIP Router V6 migration: update Link component

## Intro
As part of the [React Router V6 migration](https://github.com/sourcegraph/sourcegraph/issues/33834), we want to gradually update our codebase to stop using outdated APIs. 

These include Router V5 internals and direct use of `history`, so that later we can remove these dependencies. 
The approach we're taking roughly follows the [official V5 → V6 migration guide](https://github.com/remix-run/react-router/discussions/8753).

## This PR description
This PR does two things.

1. Wraps all Routes in `CompatRoute` as suggested in the migration guide. This allows us to use both old and new APIs simultaneously - and therefore migrate incrementally.
2. Updates the `AnchorLink` component, which is one of the most fundamental to the app navigation and the deepest leaf in the component tree.

## Test plan

1. Check out the branch
2. Check carefully that links in the app work (TBD) 

## App preview:

- [Web](https://sg-web-og-react-router-migration-link.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-aetkbfcnuu.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
